syntax = "proto3";
package collection_config;




import "google/protobuf/struct.proto";

/*
ListCollectionConfig请求
*/
message ListCollectionConfigRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 pageSize = 2 ;
    /*
    是否启禁用
    */
    int32 disabled = 3 ;
    /*
    是否全部
    */
    int32 isAll = 4 ;
}

/*
ListCollectionConfig返回
*/
message ListCollectionConfigResponse {
message List {
message Kwargs {
    /*
    键
    */
    string key = 1 ;
    /*
    值
    */
    google.protobuf.Value value = 2 ;
}
        
message Env {
    /*
    键
    */
    string key = 1 ;
    /*
    值
    */
    google.protobuf.Value value = 2 ;
}
        
message Script {
    /*
    脚本Id
    */
    string id = 1 ;
    /*
    脚本版本
    */
    string version = 2 ;
    /*
    脚本类型
    */
    string type = 3 ;
    /*
    脚本内容
    */
    string content = 4 ;
    /*
    脚本名称
    */
    string name = 5 ;
}
        
message HostRange {
message Query {
    /*
    键
    */
    string key = 1 ;
    /*
    操作符
    */
    string op = 2 ;
    /*
    值
    */
    google.protobuf.Value value = 3 ;
}
        
    
    /*
    模型ID
    */
    string objectId = 1 ;
    /*
    获取主机Key路径
    */
    string hostKey = 2 ;
    /*
    查询者
    */
    string querier = 3 ;
    /*
    查询类型
    */
    string type = 4 ;
    /*
    查询策略字符串
    */
    string queryString = 5 ;
    /*
    查询策略ID
    */
    string queryId = 6 ;
    /*
    查询策略ID
    */
    repeated Query query = 7 ;
}
        
    
    /*
    ORG
    */
    int32 org = 1 ;
    /*
    ID
    */
    string id = 2 ;
    /*
    名称
    */
    string name = 3 ;
    /*
    参数
    */
    repeated Kwargs kwargs = 4 ;
    /*
    超时时间
    */
    int32 timeout = 5 ;
    /*
    环境变量
    */
    repeated Env env = 6 ;
    /*
    是否禁用
    */
    bool disabled = 7 ;
    /*
    标签
    */
    repeated string labels = 8 ;
    /*
    脚本
    */
    Script script = 9 ;
    /*
    是否忽略不符合要求数据
    */
    bool ignoreInvalid = 10 ;
    /*
    主机范围
    */
    HostRange hostRange = 11 ;
    /*
    采集间隔
    */
    int32 interval = 12 ;
    /*
    缓存过期次数（时间）
    */
    int32 cacheTtl = 13 ;
    /*
    执行时间范围（24小时）
    */
    string timeRange = 14 ;
    /*
    采集类型ID
    */
    string clazzId = 15 ;
    /*
    创建者
    */
    string creator = 16 ;
    /*
    修改者
    */
    string modifier = 17 ;
    /*
    创建时间
    */
    int32 ctime = 18 ;
    /*
    修改时间
    */
    int32 mtime = 19 ;
    /*
    关联模型ID
    */
    string objectId = 20 ;
    /*
    关联模型实例ID宏
    */
    string instanceIdMacro = 21 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListCollectionConfigApi返回
*/
message ListCollectionConfigResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    collection_config.ListCollectionConfigResponse data = 4 ;
}

